From 8db87cb722393f86d155da7620fa6c15ea8f1259 Mon Sep 17 00:00:00 2001
From: Eugen Hristev <eugen.hristev@microchip.com>
Date: Tue, 21 May 2019 12:36:46 +0300
Subject: [PATCH] make QGraphicsItem::update virtual

This is used to prevent update() calls from happening on some QGraphicsItem
objects.

Signed-off-by: Joshua Henderson <joshua.henderson@microchip.com>
[eugen.hristev@microchip.com]: ported on QT 5.12.2
Signed-off-by: Eugen Hristev <eugen.hristev@microchip.com>
---
 src/widgets/graphicsview/qgraphicsitem.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/widgets/graphicsview/qgraphicsitem.h b/src/widgets/graphicsview/qgraphicsitem.h
index c228e765d8..9a118878a3 100644
--- a/src/widgets/graphicsview/qgraphicsitem.h
+++ b/src/widgets/graphicsview/qgraphicsitem.h
@@ -342,7 +342,7 @@ public:
 
     // Drawing
     virtual void paint(QPainter *painter, const QStyleOptionGraphicsItem *option, QWidget *widget = nullptr) = 0;
-    void update(const QRectF &rect = QRectF());
+    virtual void update(const QRectF &rect = QRectF());
     inline void update(qreal x, qreal y, qreal width, qreal height);
     void scroll(qreal dx, qreal dy, const QRectF &rect = QRectF());
 
-- 
2.21.0

